src/VolleyManagement.API/Controllers/AccountController.cs

```diff
@@ -0,0 +1,123 @@
+            IUserService userService,
+            ISecretsProvider configuration)
+        {
+            _userManager = userManager;
+            _userService = userService;
+            _configuration = configuration;
+        }
+
+        /// <summary>
+        /// Login user
+        /// </summary>
+        /// <param name="user">json with user info provided by UI</param>
+        /// <returns>Action result</returns>
+        [HttpPost]
+        [AllowAnonymous]
+        public async Task<IActionResult> TokenSignin(string user)
+        {
+            var userInfoFromWebApp = JsonConvert.DeserializeObject<GoogleUserInfo>(user);
+
+            var validatedLoginInfoFromGoogle = await GoogleJsonWebSignature.ValidateAsync(userInfoFromWebApp.IdToken,
+                new GoogleJsonWebSignature.ValidationSettings() {
+                    Audience = new List<string> { _configuration.GetGoogleClientId().Value }
+                });
+
+            var userInSystem = await _userManager.FindByEmailAsync(validatedLoginInfoFromGoogle.Email);
+
+            if (userInSystem == null)
+            {
+                userInSystem = new UserModel {
+                    Email = validatedLoginInfoFromGoogle.Email,
+                    UserName = validatedLoginInfoFromGoogle.Email,
+                    PersonName = $"{validatedLoginInfoFromGoogle.FamilyName} {validatedLoginInfoFromGoogle.GivenName}"
+                };
+                var result = await _userManager.CreateAsync(userInSystem);
+                if (!result.Succeeded)
+                {
+                    return Unauthorized();
+                }
+                else
+                {
+                    result = await _userManager.AddLoginAsync(userInSystem.Id,
+                        new UserLoginInfo(GOOGLE, userInSystem.Email));
+                    if (!result.Succeeded)
+                    {
+                        return Unauthorized();
+                    }
+                }
+            }
+
+            var ident = await _userManager.CreateIdentityAsync(
+                userInSystem,
+                DefaultAuthenticationTypes.ApplicationCookie);
+
+            if (IsBlocked(ident.GetUserId<int>()))
+            {
+                return Unauthorized();
+            }
+
+            ident.AddClaims(
+                new List<Claim>()
+                {
+                    new Claim(nameof(validatedLoginInfoFromGoogle.Name), validatedLoginInfoFromGoogle.Email),
+                    new Claim(nameof(validatedLoginInfoFromGoogle.FamilyName), validatedLoginInfoFromGoogle.FamilyName),
+                    new Claim(nameof(validatedLoginInfoFromGoogle.GivenName), validatedLoginInfoFromGoogle.GivenName)
+                });
+
+
+            await HttpContext.SignOutAsync("VMCookie");
+            await HttpContext.SignInAsync("VMCookie",
+                new ClaimsPrincipal(ident),
+                new AuthenticationProperties { IsPersistent = false }
```

Comment:
what IsPersistent flag does?

Reply:
Gets or sets whether the authentication session is persisted across multiple requests.